File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    answer = []
    
    score = [0,0,0]
    one = [1,2,3,4,5]
    two = [2,1,2,3,2,4,2,5]
    thr = [3,3,1,1,2,2,4,4,5,5]
    len_a = len(answers)
    
    for i in range(len_a) :
        a = answers[i]
        
        if one[i%5] == a :
            score[0] += 1
        if two[i%8] == a :
            score[1] += 1
        if thr[i%10] == a :
            score[2] += 1
    
    max_s = max(score)
    for i in range(3) :
        if score[i] == max_s :
            answer.append(i+1)
    
    return answer

File: 프로그래머스/1/42840. 모의고사/모의고사.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> answers) {
    vector<int> answer;
    
    vector<int> supo1 = {1, 2, 3, 4, 5};
    vector<int> supo2 = {2, 1, 2, 3, 2, 4, 2, 5};
    vector<int> supo3 = {3, 3, 1, 1, 2, 2, 4, 4, 5, 5};
    
    int idx1 = 0;
    int idx2 = 0;
    int idx3 = 0;
    
    int collect1 = 0;
    int collect2 = 0;
    int collect3 = 0;
    
    for (int a : answers) {
        int value1 = supo1[idx1];
        int value2 = supo2[idx2];
        int value3 = supo3[idx3];
        
        if (value1 == a) {
            collect1 += 1;
        } 
        if (value2 == a) {
            collect2 += 1;
        }
        if (value3 == a) {
            collect3 += 1;
        }
        
        idx1 += 1;
        idx2 += 1;
        idx3 += 1;
        
        if (idx1 == supo1.size()) {
            idx1 = 0;
        }
        if (idx2 == supo2.size()) {
            idx2 = 0;
        }
        if (idx3 == supo3.size()) {
            idx3 = 0;
        }
    } 
    
    int max_collect = max({collect1, collect2, collect3});
    if (collect1 == max_collect) {
        answer.push_back(1);
    }
    if (collect2 == max_collect) {
        answer.push_back(2);
    }
    if (collect3 == max_collect) {
        answer.push_back(3);
    }
    
    return answer;
}